client/src/pages/Solutions.tsx

```diff
@@ -1,138 +1,82 @@
-import { useState, useEffect } from "react";
-import { Container, Row, Col, Form } from "react-bootstrap";
-import SolutionCard from "../components/SolutionCard.tsx";
-import { SolutionData } from "../types/SolutionData.ts";
-import MasonryGrid from "../components/MasonryGrid.tsx";
+import React, { useState, useEffect } from "react";
+import { Container, Row, Col, Alert, Spinner, ListGroup } from "react-bootstrap";
+import { useParams, Link } from "react-router-dom";
+import useCheckRole from "../hooks/useCheckRole";
 
-const Solutions = () => {
-  const [solutions, setSolutions] = useState<SolutionData[]>([]);
-  const [showingSolutions, setShowingSolutions] = useState<SolutionData[]>([]);
-  const [challengeNames, setChallengeNames] = useState<string[]>([]);
-  const [solvedChallenges, setSolvedChallenges] = useState<string[]>([]);
-  const [thisUser, setThisUser] = useState<{username: string, role: string}>({username: "", role: ""});
+interface SolutionSummary {
+  id: string;
+  username: string;
+  title: string;
+  submittedAt: string;
+}
 
+const Solutions: React.FC = () => {
+  const { questionId } = useParams<{ questionId: string }>();
+  const [solutions, setSolutions] = useState<SolutionSummary[]>([]);
```

Comment:
This is much better. But I'm curious as to why you merge the states here but distribute them in the submit solution form?